subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppExecutablePlugin.java

```diff
@@ -118,26 +140,51 @@ public void execute(MavenPublication publication) {
                                 publication.from(mainVariant);
                             }
                         });
-                        extension.getPublications().create("debug", MavenPublication.class, new Action<MavenPublication>() {
+                        extension.getPublications().create("debugShared", MavenPublication.class, new Action<MavenPublication>() {
```

Comment:
We don't want separate modules for (debug, release) x (shared, static) combination. I think we want a debug module that contains all the debug binaries, and a release module that contains all the release binaries.